File: src/main/java/no/nav/domain/pensjon/regler/beregning2011/FremskrevetAfpLivsvarig.java
package no.nav.domain.pensjon.regler.beregning2011;

import no.nav.domain.pensjon.regler.kode.YtelsekomponentTypeCti;

public class FremskrevetAfpLivsvarig extends AfpLivsvarig {
    

    private double reguleringsfaktor;
    private int gap;
    private double gjennomsnittligUttaksgradSisteAr;

    public FremskrevetAfpLivsvarig() {
        super();
        ytelsekomponentType = new YtelsekomponentTypeCti("FREM_AFP_LIVSVARIG");
    }


}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning2011/FremskrevetAfpLivsvarig.kt
package no.nav.pensjon.regler.domain.beregning2011

import no.nav.pensjon.regler.domain.enum.YtelseskomponentTypeEnum

class FremskrevetAfpLivsvarig : AbstraktAfpLivsvarig(), Regulering {
    override var reguleringsfaktor = 0.0
    override var gap = 0
    var gjennomsnittligUttaksgradSisteAr = 0.0

    override var ytelsekomponentTypeEnum: YtelseskomponentTypeEnum = YtelseskomponentTypeEnum.FREM_AFP_LIVSVARIG
}